### PR TITLE
ajm/1266 pugixml fix

### LINKED ISSUE #1261: Rogue pugixml-related files being installed
**Describe the bug**
It has been noticed when we create a carta-backend RPM (and carta-backend Debian package), it makes and tries to install a bunch of pugixml related files. This should not happen because we only use pugixml as a third-party static dependency during build time. It should not be installing any files on the system.

For reference, the following is the error we get at the end of the rpmbuild stage listing the files in question:
```
Checking for unpackaged file(s): /usr/lib/rpm/check-files /root/rpmbuild/BUILDROOT/carta-backend-beta-4.0+2023.5.4-1.x86_64
error: Installed (but unpackaged) file(s) found:
   /usr/include/pugiconfig.hpp
   /usr/include/pugixml.hpp
   /usr/lib64/cmake/pugixml/pugixml-config-version.cmake
   /usr/lib64/cmake/pugixml/pugixml-config.cmake
   /usr/lib64/cmake/pugixml/pugixml-targets-release.cmake
   /usr/lib64/cmake/pugixml/pugixml-targets.cmake
   /usr/lib64/libpugixml.a
   /usr/lib64/pkgconfig/pugixml.pc
```
This problem was introduced when we began using pugixml as a third-party library in [PR#1288](https://github.com/CARTAvis/carta-backend/pull/1228)

**To Reproduce**
(I haven't checked the following method, but I think it may work)
1. Build a carta-backend
2. Check the time stamp on`/usr/lib64/libpugixml.a`
3. It may indicate that the file was freshly installed.

**Expected behavior**
During the carta-backend make process, the third-party pugixml library should not install any files on your system.

**Platform info (please complete the following information):**
 - OS [e.g. macOS Monterey]: CentOS7
 - Browser [e.g. chrome, safari, electron app]: not applicable 
 - Browser version [e.g. 22]: not applicable
 - Backend branch [e.g. dev, v3b2 release]: v4.0.0-beta.1
 - Frontend branch [e.g. dev, v3b2 release]: not applicable

### PR DESCRIPTION
**Pull request by ajm-ska**
**Description**

This addresses issue #1261 

It uses @confluence's clever suggestion to simply include the pugixml cpp file directly in our build. It seems to work! I have attached 3 images as a demonstration. In each case, I open a catalog file.

1. The current v4.0.0-beta.1
v4beta1

2. New method without the `-DPUGIXML_COMPACT` flag
new_no_PUGIXML_COMPACT

3. New method with the `-DPUGIXML_COMPACT` flag
new_PUGIXML_COMPACT

We can see that the new method works, and it is clearly built using the compact mode because the RAM usage in plot 3 is practically identical to plot 1.

The new method also solves the pugixml install file issue:

In the old v4.0.0-beta.1 branch, when we do `make install` it installs pugixml files on your system that it should not install:
```
root@afe418483e39:/mount/carta-backend/build# make install
[  1%] Built target pugixml-static
[  8%] Built target uSockets
[ 59%] Built target carta-protobuf
[ 95%] Built target carta_backend
[100%] Built target spdlog
Install the project...
-- Install configuration: "RelWithDebInfo"
-- Installing: /usr/local/bin/carta_backend
-- Set runtime path of "/usr/local/bin/carta_backend" to ""
-- Up-to-date: /usr/local/bin/carta
-- Up-to-date: /usr/local/share/carta/default.fits
-- Installing: /usr/local/lib/libpugixml.a
-- Installing: /usr/local/lib/cmake/pugixml/pugixml-targets.cmake
-- Installing: /usr/local/lib/cmake/pugixml/pugixml-targets-relwithdebinfo.cmake
-- Installing: /usr/local/lib/cmake/pugixml/pugixml-config-version.cmake
-- Installing: /usr/local/lib/cmake/pugixml/pugixml-config.cmake
-- Installing: /usr/local/lib/pkgconfig/pugixml.pc
-- Up-to-date: /usr/local/include/pugiconfig.hpp
-- Up-to-date: /usr/local/include/pugixml.hpp
```

In the new branch, when we do `make install`, it now only installs the expected files:
```
root@150e535df8fd:/mount/carta-backend/build# make install
[ 34%] Built target carta-protobuf
[ 39%] Built target uSockets
[ 63%] Built target carta_backend
[ 66%] Built target spdlog
[100%] Built target carta_backend_tests
Install the project...
-- Install configuration: "RelWithDebInfo"
-- Installing: /usr/local/bin/carta_backend
-- Set runtime path of "/usr/local/bin/carta_backend" to ""
-- Installing: /usr/local/bin/carta
-- Installing: /usr/local/share/carta/default.fits
```

I also removed the installation of the pugixml packages from the Dockerfiles because they are no longer necessary. 

Changelog not updated yet. Do we add it to the "Fixed" section under [4.0.0-beta.1], or to a new [unreleased] "Fixed" section?

**Checklist**

- [x] changelog updated
- [ ] e2e test passing / added corresponding fix
- [x] no protobuf update needed
- [x] added reviewers and assignee
- [x] added ZenHub estimate, milestone, and release

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -155,9 +155,7 @@ include_directories(${CMAKE_SOURCE_DIR}/third-party/include)
 install_uWebSockets()
 
 #pugixml
-SET(PUGIXML_COMPACT ON CACHE BOOL "" FORCE)
```
Comment (confluence) [inline CMakeLists.txt:158]: We still need to set this option by default. I think that the same configuration line will work, but please test this.

**Review by confluence — CHANGES_REQUESTED** [review]
I think that the changelog entry should go in the new unreleased changelog (fixing the issue in the packages doesn't count).

File: CMakeLists.txt (hunk 1 above)
Comment (ajm-ska) [reply CMakeLists.txt:158]: I'm sorry, maybe I got files mixed up here! 
I thought I had added  `-DPUGIXML_COMPACT` to line 251
```
add_definitions(-DHAVE_HDF5 -DPUGIXML_COMPACT)
```
I better wait until the morning.

Comment (ajm-ska) [reply CMakeLists.txt:158]: Oh yes, I must have removed it when testing scenario 2, and then made a different commit and didn't add it back.
I believe `-DPUGIXML_COMPACT` does the same as `SET(PUGIXML_COMPACT ON CACHE BOOL "" FORCE)`, in fact, I think it is the way we need to do it now.

Comment (ajm-ska) [reply CMakeLists.txt:158]: I'm sorry, now I made another mistake and added both those options in!
I definitely better wait until tomorrow before I make any more mistakes. Then I'll do a performance test on both scenarios.

**Comment by ajm-ska** [comment]
@confluence I'm sorry about the mistakes last night! I have done two further quick tests.

1. With `SET(PUGIXML_COMPACT ON CACHE BOOL "" FORCE)` only:
old

2. With `-DPUGIXML_COMPACT)` only:
new

The tests show that when we include the pugixml.cpp file directly into our build, the only way to enable the "compact" mode is to use the `-DPUGIXML_COMPACT` flag. 

Now I have `-DPUGIXML_COMPACT` added in the committed CMakeLists.txt.

**Review by confluence — APPROVED** [review]

**Review by kswang1029 — APPROVED** [review]
This looks good. Tested and see identical behavior before/after. No regression from e2e tests.

Please do update dev first before merging.

**Pull request merged**